paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetLoader.kt

```diff
@@ -148,83 +110,58 @@ internal class DefaultCustomerSheetLoader(
+                    }?.let {
+                        PaymentSelection.Saved(it)
+                    }
+                }
+                is SavedSelection.None -> null
             }
-        }.map {
-            val paymentMethods = it.first
-            val paymentOption = it.second
-            val selection = paymentOption?.toPaymentSelection { id ->
-                paymentMethods.find { it.id == id }
-            }
-            Pair(paymentMethods, selection)
-        }.fold(
-            onSuccess = { result ->
-                val paymentSelection = result.second
-
-                val sortedPaymentMethods = sortPaymentMethods(
-                    paymentMethods = result.first,
-                    selection = paymentSelection as? PaymentSelection.Saved
-                )
+        }
 
-                val elementsSession = elementsSessionWithMetadata.elementsSession
-                val metadata = elementsSessionWithMetadata.metadata
+        val sortedPaymentMethods = sortPaymentMethods(
+            paymentMethods = customerSheetSession.paymentMethods,
+            selection = paymentSelection as? PaymentSelection.Saved
+        )
 
-                val supportedPaymentMethods = metadata.sortedSupportedPaymentMethods()
+        val supportedPaymentMethods = metadata.sortedSupportedPaymentMethods()
 
-                val validSupportedPaymentMethods = filterSupportedPaymentMethods(supportedPaymentMethods)
+        val validSupportedPaymentMethods = filterSupportedPaymentMethods(supportedPaymentMethods)
 
-                Result.success(
-                    CustomerSheetState.Full(
-                        config = configuration,
-                        paymentMethodMetadata = metadata,
-                        supportedPaymentMethods = validSupportedPaymentMethods,
-                        customerPaymentMethods = sortedPaymentMethods,
-                        paymentSelection = paymentSelection,
-                        validationError = elementsSession.stripeIntent.validate(),
-                        customerPermissions = CustomerPermissions(
-                            // Should always be true for `legacy` case
-                            canRemovePaymentMethods = true,
-                        )
-                    )
-                )
-            },
-            onFailure = { cause, _ ->
-                Result.failure(cause)
-            }
+        return CustomerSheetState.Full(
+            config = configuration,
+            paymentMethodMetadata = metadata,
+            supportedPaymentMethods = validSupportedPaymentMethods,
+            customerPaymentMethods = sortedPaymentMethods,
+            paymentSelection = paymentSelection,
+            validationError = customerSheetSession.elementsSession.stripeIntent.validate(),
+            customerPermissions = CustomerPermissions(
+                // Should always be true for `legacy` case
+                canRemovePaymentMethods = true,
```

Comment:
by "legacy", does this mean non-customer sessions? If so, should this logic be moved out of here and into the customer adapter vs. customer sessions implementations?

Reply:
Ah this was moved but I didn't update the usage here. Will do that now.